File: backend/ZJAdmin.Api/Controllers/AuthController.cs
using System.Security.Claims;
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using ZJAdmin.Api.DTOs;
using ZJAdmin.Api.Services;

namespace ZJAdmin.Api.Controllers;

[ApiController]
[Route("api/v1/auth")]
public class AuthController : ControllerBase
{
    private readonly AuthService _authService;
    private readonly UserService _userService;

    public AuthController(AuthService authService, UserService userService)
    {
        _authService = authService;
        _userService = userService;
    }

    [HttpPost("login")]
    [AllowAnonymous]
    public async Task<ApiResponse> Login([FromBody] LoginRequest request)
    {
        return await _authService.Login(request);
    }

    [HttpPost("logout")]
    public ApiResponse Logout()
    {
        return ApiResponse.Success(null, "退出成功");
    }

    [HttpGet("profile")]
    public async Task<ApiResponse> GetProfile()
    {
        var userId = GetUserId();
        var user = await _authService.GetCurrentUser(userId);
        if (user == null)
            return ApiResponse.Error("用户不存在", 404);
        return ApiResponse.Success(user);
    }

    [HttpPut("profile")]
    public async Task<ApiResponse> UpdateProfile([FromBody] UpdateProfileRequest request)
    {
        var userId = GetUserId();
        return await _userService.UpdateProfile(userId, request);
    }

    [HttpPut("password")]
    public async Task<ApiResponse> ChangePassword([FromBody] ChangePasswordRequest request)
    {
        var userId = GetUserId();
        return await _userService.ChangePassword(userId, request);
    }

    private long GetUserId()
    {
        var claim = User.FindFirst(ClaimTypes.NameIdentifier);
        return long.Parse(claim?.Value ?? "0");
    }
}

File: backend-java/src/main/java/com/zjadmin/controller/AuthController.java
package com.zjadmin.controller;

import com.zjadmin.dto.*;
import com.zjadmin.security.SecurityUtils;
import com.zjadmin.service.AuthService;
import com.zjadmin.service.UserService;
import jakarta.validation.Valid;
import lombok.RequiredArgsConstructor;
import org.springframework.security.crypto.password.PasswordEncoder;
import org.springframework.web.bind.annotation.*;

import java.util.List;

@RestController
@RequestMapping("/api/v1/auth")
@RequiredArgsConstructor
public class AuthController {

    private final AuthService authService;
    private final UserService userService;
    private final PasswordEncoder passwordEncoder;

    @PostMapping("/login")
    public ApiResponse<LoginResponse> login(@Valid @RequestBody LoginRequest request) {
        return authService.login(request);
    }

    @PostMapping("/logout")
    public ApiResponse<?> logout() {
        return ApiResponse.success(null, "Logout success");
    }

    @GetMapping("/menus")
    public ApiResponse<List<PermissionTreeNode>> getMenus() {
        Long userId = SecurityUtils.getCurrentUserId();
        List<PermissionTreeNode> menus = authService.getCurrentUserMenus(userId);
        return ApiResponse.success(menus);
    }

    @GetMapping("/profile")
    public ApiResponse<UserInfo> getProfile() {
        Long userId = SecurityUtils.getCurrentUserId();
        UserInfo user = authService.getCurrentUser(userId);
        if (user == null) return ApiResponse.error("User not found", 404);
        return ApiResponse.success(user);
    }

    @PutMapping("/profile")
    public ApiResponse<?> updateProfile(@RequestBody UpdateProfileRequest request) {
        Long userId = SecurityUtils.getCurrentUserId();
        return userService.updateProfile(userId, request);
    }

    @PutMapping("/password")
    public ApiResponse<?> changePassword(@Valid @RequestBody ChangePasswordRequest request) {
        Long userId = SecurityUtils.getCurrentUserId();
        return userService.changePassword(userId, request);
    }

    @GetMapping("/hash")
    public ApiResponse<String> generateHash(@RequestParam String password) {
        return ApiResponse.success(passwordEncoder.encode(password));
    }
}